2_process/src/munge_inst_timeseries.R

```diff
@@ -55,3 +55,31 @@ aggregate_data_to_hourly <- function(inst_data,output_tz){
 
 }
 
+aggregate_data_to_daily <- function(inst_data, daily_data){
+  #' 
+  #' @description Function to aggregate instantaneous NWIS data collected at sub-hourly (e.g. 15/5/30 min) intervals to hourly min/mean/maxs
+  #'
+  #' @param inst_data a data frame containing the downloaded time series for NWIS instantaneous site. 
+  #' inst_data must include the following columns: c("Value_Inst","dateTime","site_no")
+  #' @param daily_data a data frame the downloaded daily time series of DO data. This is used so that if a site is already in 
+  #'  the daily sites, we won't do the aggregating here
+  #'
+  #' @value A data frame containing daily min, mean, and max values the parameter of interest
+  
+  
+  only_inst_data = setdiff(inst_data$site_no, daily_data$site_no)
+  
+  daily_values <- p1_inst_data %>%
+    filter(site_no %in% only_inst_data) %>%
+    mutate(Date = as.Date(dateTime, format="%Y-%m-%d")) %>%
+    group_by(Date, site_no) %>%
+    summarise(Value = mean(Value_Inst), 
+              Value_Min = min(Value_Inst), 
+              Value_Max = max(Value_Inst), 
+              na_count=sum(is.na(Value_Inst)), 
+              value_count=sum(!is.na(Value_Inst))) %>%
+    mutate(percent_coverage=value_count/(value_count + na_count)) %>%
+    filter(percent_coverage > 0.5)
```

Comment:
Does this look like the right way to be filtering these? And if so, is a 50% daily coverage what we want do go with?

Reply:
>Does this look like the right way to be filtering these? 

btw, I did a spot check on this and it was what I was expecting, but I just want to make sure I didn't overlook anything simple

Reply:
The code looks looks good to me!

Reply:
Doh! @jsadler2 I missed something simple in the lines above this code snipped. We need to add `na.rm=TRUE` as an argument to the `min`, `max`, and `mean` functions. Otherwise, if there are any NA values for a uv interval, the daily summary will return NA.

Reply:
Thanks for catching that! It's been adjusted now.

Reply:
Could make the % cutoff a function argument, but 50% as a threshold sounds good. It does make me wonder if we should require a certain number of samples, too. If we had 2 samples, they would be our max and min values.  

For something like DO, it could help to report the time of the max and min.

Reply:
Time is not reported in the dv data (`p1_daily_data`) for salinity, but I wouldn't expect time of day to be important for inland salinity. For tides it could be

Reply:
> Could make the % cutoff a function argument

I agree that the percent coverage cutoff should be a function argument since the USGS guidance varies among some constituents (e.g. for DO, it is 25%). 

> For something like DO, it could help to report the time of the max and min.

I agree that knowing the timing of daily min/max may help inform the processes that are driving daily variation in DO. We would want to know that information for all instantaneous sites rather than just the subset of "inst only" sites, so if we didn't want to include here, we could either create a separate target, or add timing in later when we are working on the dynamic models.

Reply:
I really like the idea of including this information. I've played around a little bit to calculate these, but haven't found a super clean solution. Given that, and since we have no immediate plans to use this information, I suggest that I open a new issue for this so we don't forget about it. That way we can move forward with merging what we have for calculating the min/mean/maxs and leave the calculating of the min/max timings to another PR. Does that sound good, @lekoenig, @jds485 ?

Reply:
OK, glad we're all in agreement that the timing data could potentially be useful (and thanks for taking a first stab at it, @jsadler2!). I'm fine with moving forward and leaving the timing for another PR. I have some ideas to try and could look into this further if we have an issue for it.

Reply:
Sounds good to me to move timing data to a new issue

Reply:
👍 - here it is: https://github.com/USGS-R/drb-inland-salinity-ml/issues/34